jgrapht-core/src/main/java/org/jgrapht/alg/Clique.java

```diff
@@ -0,0 +1,88 @@
+package org.jgrapht.alg;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.jgrapht.Graphs;
+import org.jgrapht.UndirectedGraph;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.SimpleGraph;
+
+
+/**
+ * Finds a maximum (weight) clique in a undirected graph. A clique
+ * for graph G = (V, E) with weight function w: V -> |R_+ is a subset of vertices 
+ * V', such that all vertices in V' are pairwise connected via an edge in G. 
+ * The weight of a clique V' is the sum of all w(v), v ∈ V'.
+ * 
+ * The implementation relies on the fact that if V is a maximum weight independent
+ * set for graph G = (V,E) with weight function w: V -> |R_+, then V is a 
+ * a maximum weight clique in the complement of G. The complement of a graph G
+ * is the graph G' = (V',E'), such that for all u,v ∈ V
+ * 		
+ * 		{u,v} ∈ E <=> not {u,v} ∈ E'
+ *
+ * @author Nils Olberg
+ */
+
+public abstract class Clique {
```

Comment:
There's already one clique finder implementation (BronKerboschCliqueFinder), and now this is a second one.  So we should try to rationalize the interface/package/naming.

The pattern we've started following is to add an interface in the interfaces package, then create a new subpackage (e.g. clique) to contain the implementation classes (and "moving" the existing one via deprecation).

So it would be something like interface CliqueFinder, with implementations BronKerboschCFImpl and IndependentSetComplementCFImpl.

@jkinable what do you think?